deepnote_toolkit/sql/sql_execution.py

```
@@ -517,12 +519,78 @@ def _query_data_source(
+                dbapi_conn, self._self_cursors
+            )
+        except Exception as e:
+            logger.warning(f"Could not install DBAPI wrapper: {e}")
+
+    def cancel_all_cursors(self):
+        """Cancel all tracked cursors. Best-effort, ignores errors."""
+        for cursor in list(self._self_cursors):
+            _cancel_cursor(cursor)
+
+
+def _cancel_cursor(cursor):
```

Ruff (0.14.14)

[warning] Missing return type annotation for private function `_cancel_cursor`

Add return type annotation: `None`

(ANN202)